magefile.go

```diff
@@ -654,7 +654,7 @@ func packageAgent(requiredPackages []string, packagingFn func()) {
 			dependencies := []string{
 				"auditbeat", "filebeat", "heartbeat", "metricbeat", "osquerybeat", "packetbeat", // beat dependencies
 				"apm-server",
-				// "cloudbeat", // TODO: add once working
+				"cloudbeat", // only supporting linux/amd64 or linux/arm64
```

Comment:
We need to conditionally include `cloudbeat` only for the supported platforms, otherwise the `fetchBinaryFromArtifactsApi()` call below fails because there are no artefacts published for these artifacts.

Running `DEV=true EXTERNAL=true SNAPSHOT=true PLATFORMS=darwin/arm64 PACKAGES=tar.gz mage -v package` fails for me on my M1 Mac. It looks like APM server has the same problem because they don't publish ARM artifacts.

Reply:
I added a way to deal with this in https://github.com/elastic/elastic-agent/pull/1908. You can just rebase on that change and exclude every platform except the two listed here.

Reply:
That's a valid point, I should have definitely considered it. See my comments on https://github.com/elastic/elastic-agent/pull/1908